app/src/main/java/org/akvo/flow/activity/SurveyActivity.java

```diff
@@ -127,12 +127,14 @@ public void onDrawerOpened(View drawerView) {
         }
 
         // Start the setup Activity if necessary.
+        boolean noDevIdYet = false;
         if (!Prefs.getBoolean(this, Prefs.KEY_SETUP, false)) {
+            noDevIdYet = true;
             startActivityForResult(new Intent(this, AddUserActivity.class), REQUEST_ADD_USER);
         }
 
         displaySelectedUser();
-        startServices();
+        startServices(noDevIdYet);
```

Comment:
Yes, that is the idea. It will not be started here, but only on return from the get-user-and-devid screen.